deploy/crds/openebs.io_blockdevices_crd.yaml

```diff
@@ -0,0 +1,247 @@
+                      For example, if the object reference is to a container within
+                      a pod, this would take on a value like: "spec.containers{name}"
+                      (where "name" refers to the name of the container that triggered
+                      the event) or if no container name is specified "spec.containers[2]"
+                      (container with index 2 in this pod). This syntax is chosen
+                      only to have some well-defined way of referencing a part of
+                      an object. TODO: this design is not final and this field is
+                      subject to change in the future.'
+                    type: string
+                  kind:
+                    description: 'Kind of the referent. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+                    type: string
+                  name:
+                    description: 'Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names'
+                    type: string
+                  namespace:
+                    description: 'Namespace of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/'
+                    type: string
+                  resourceVersion:
+                    description: 'Specific resourceVersion to which this reference
+                      is made, if any. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#concurrency-control-and-consistency'
+                    type: string
+                  uid:
+                    description: 'UID of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#uids'
+                    type: string
+                type: object
+              details:
+                description: Details contain static attributes of BD like model,serial,
+                  and so forth
+                properties:
+                  compliance:
+                    description: Compliance is standards/specifications version implemented
+                      by device firmware  such as SPC-1, SPC-2, etc
+                    type: string
+                  deviceType:
+                    description: DeviceType represents the type of device like sparse,
+                      disk, partition, lvm, raid
+                    enum:
+                    - disk
+                    - partition
+                    - sparse
+                    - loop
+                    - lvm
+                    - raid
+                    type: string
+                  driveType:
+                    description: DriveType is the type of backing drive, HDD/SSD
+                    enum:
+                    - HDD
+                    - SDD
+                    - Unknown
```

Comment:
Should we consider this Unknown value, Is it similar to `""`?

Reply:
Since we have given this field as a mandatory field, need to check if we should give this. Or we can remove it from mandatory field.

Reply:
`Unknown` is a string itself. Just thought that if we couldn't detect the drive type, we can mark it as unknown in the CRD.  
Yep, we can remove it from mandatory field

Reply:
Still hitting error of not accepting `""` empty value. I will make a change in code to change empty value to `Unknown`. But still I think we need to add `""` also here, otherwise the following case wont work.

1. Already a BD exists with empty value for driveType. And the users is trying to do some update operation like removing finalizer. This edit wont work, because the validation will fail for the `driveType` field.

Reply:
@harshthakur9030 Can you add empty value also here.